### PR TITLE
fix: set Explore defaults on app load

### PR DESCRIPTION
**Pull request by SgtPooki**
fix: set Explore defaults on app load

* sets localStorage settings required by ipld-explorer-components on app load
    * disable remote gateways
    * enable local gateways

### CODE REVIEW

**Comment by SgtPooki** [comment]
note that we need to update ipld-explorer-components to use @helia/block-brokers@3.0.4 so we do not need to set `allowInsecure: true`


```console
╰─ ✘ 1 ❯ npm why @helia/block-brokers
@helia/block-brokers@3.0.3
node_modules/@helia/block-brokers
  @helia/block-brokers@"^3.0.3" from @helia/http@1.0.10
  node_modules/@helia/http
    @helia/http@"^1.0.10" from ipld-explorer-components@7.0.2
    node_modules/ipld-explorer-components
      ipld-explorer-components@"^7.0.2" from the root project
  @helia/block-brokers@"^3.0.3" from ipld-explorer-components@7.0.2
  node_modules/ipld-explorer-components
    ipld-explorer-components@"^7.0.2" from the root project
```

File: src/bundles/ipfs-provider.js [hunk 1]
```diff
@@ -315,6 +315,25 @@ const selectors = {
...
+    const useRemoteGatewaysToExplore = localStorage.getItem('explore.ipld.gatewayEnabled')
+    if (useRemoteGatewaysToExplore === null) {
+      // by default, disable remote gateways for the Explore page (i.e. ipld-explorer-components)
+      await writeSetting('explore.ipld.gatewayEnabled', false)
+    }
+
+    const kuboGateway = readSetting('kuboGateway')
+    if (kuboGateway === null || typeof kuboGateway === 'string' || typeof kuboGateway === 'boolean' || typeof kuboGateway === 'number') {
+      // empty or invalid, set defaults
+      await writeSetting('kuboGateway', { trustlessBlockBrokerConfig: { init: { allowLocal: true, allowInsecure: true } } })
```
Comment (lidel) [inline src/bundles/ipfs-provider.js:330]: nit: shouldn't we store port and host? user may have it on something other than `127.0.0.1:8080`, for example ex-brave user could have `127.0.0.1:48080`

File: src/App.js [hunk 2]
```diff
@@ -32,6 +33,11 @@ export class App extends Component {
     isOver: PropTypes.bool.isRequired
   }
 
+  constructor (props) {
+    super(props)
+    props.doSetupLocalStorage()
```
Comment (lidel) [inline src/App.js:38]: This does not seem to work on initial page load.
If you use empty browser profile and load webui, and then go to ipld explorer and click on wikipedia example you will see spinner forever, and no request:

>  
>  

If you then force-reload the page (F5) the helia will initialize with local gateway and work fine:

>

**Review by lidel — CHANGES_REQUESTED** [review]

File: src/bundles/ipfs-provider.js (hunk 1 above)
Comment (SgtPooki) [reply src/bundles/ipfs-provider.js:330]: That happens elsewhere so i was planning to leave that alone

File: src/App.js (hunk 2 above)
Comment (SgtPooki) [reply src/App.js:38]: I think landing directly on explore page might cause the issue? 

Issue doesn't happen if you land on status page first

Comment (SgtPooki) [reply src/App.js:38]: Might need to call the localStorage fn in router / component loader

File: src/bundles/ipfs-provider.js (hunk 1 above)
Comment (SgtPooki) [reply src/bundles/ipfs-provider.js:330]: I guess that would work for now but it doesn't feel great

**Comment by SgtPooki** [comment]
@lidel try now, it should work now that we're explicitly using ipld-explorer-components@7.0.3

File: src/App.js (hunk 2 above)
Comment (SgtPooki) [reply src/App.js:38]: landing directly on explore page works for me with the latest.

https://github.com/user-attachments/assets/c7953852-56af-48e2-840b-3b41c76f83d3

**Review by SgtPooki — COMMENTED** [review]
self review

**Review by lidel — APPROVED** [review]
Thanks, it now works :partying_face: feel free to ship a patch release.

(I also confirmed running on a custom RPC port also works)

File: src/App.js (hunk 2 above)
Comment (lidel) [reply src/App.js:38]: Yep, that was it. New version works fine.

ps. TIL [Firefox does not support HEVC while it is encumbered by patents](https://www.reddit.com/r/firefox/comments/zeg2qy/firefox_doesnt_support_h265_hevc_for_patent/), so inline player does not work (but i played fine with external player): 

> image

**Pull request merged**